### ISSUE TITLE
warning in build.sh

### ISSUE THREAD

**Issue by CangHaiQingYue** [zeakey/hed#11]
Hello, another question came out,
When I `sudo bash bulid.sh` there is a warning `src/caffe/layers/balance_cross_entropy_loss_layer.cu(45): warning: statement is unreachable
`
After that, I run python train.py an mistake throw out.
`Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: BalanceCrossEntropyLoss (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Concat, ContrastiveLoss, Convolution, Crop, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, Input, LRN, LSTM, LSTMUnit, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Parameter, Pooling, Power, Python, RNN, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, Swish, TanH, Threshold, Tile, WindowData)
`
Can you help me?

**Comment by zeakey** [comment]
* **The warning**: I just call `Forward_cpu` in the `Forward_gpu` function, in which I bypass the cuda codes.
This is why the warning occurs, but it doesn't matter and you can just ignore it.

* **Unknown layer type**: This is mainly because you are using another caffe on your system, since you compile the source with `sudo` (which is strongly unrecommanded), you may not have the permission to read the compiled binaries, and then Python may search other path rather than `$HED/caffe/python`.
So I suggest you to check the path by `print(caffe.__file__)` after `import caffe`, the right one should be something like `$HED/caffe/python`.

Hope it helps.

**Comment by CangHaiQingYue** [comment]
Thanks for your reply.
I will check my steps.
